src/providers/data_provider/dp.c

```diff
@@ -216,9 +208,8 @@ static void dp_init_done(struct tevent_req *subreq)
     req = tevent_req_callback_data(subreq, struct tevent_req);
     state = tevent_req_data(req, struct dp_init_state);
 
-    ret = sbus_server_create_and_connect_recv(state->provider, subreq,
-                                              &state->provider->sbus_server,
-                                              &state->provider->sbus_conn);
+    ret = sbus_connect_private_recv(state->provider, subreq,
+                                    &state->provider->sbus_conn);
```

Comment:
No behavior change. If you fail to connect, it will be a fatal error and the process will terminate.

If existing connection is lost, it will reconnect.